Dockerfile.product

```diff
@@ -16,11 +16,15 @@ FROM registry.ci.openshift.org/ocp/builder:rhel-8-base-nodejs-openshift-4.10 AS
 ADD . .
 
 USER 0
-# extract the yarn dependencies that must be provided in the dist-git lookaside cache
-RUN tar fx yarn-offline.tar
 
 # bootstrap yarn so we can install and run the other tools.
-RUN container-entrypoint npm install ./yarn-1.9.4.tgz
+RUN container-entrypoint npm install ./artifacts/yarn-v1.22.19.tar.gz
+
+COPY $REMOTE_SOURCES $REMOTE_SOURCES_DIR
+
+# use dependencies provided by Cachito
+RUN cp -f $REMOTE_SOURCES_DIR/cachito-gomod-with-deps/app/registry-ca.pem . \
+  && cp -f $REMOTE_SOURCES_DIR/cachito-gomod-with-deps/app/frontend/{.npmrc,.yarnrc,yarn.lock} frontend/
```

Comment:
The way that Cachito works is setting up a temporary npm registry so it will replace the resolved URLs in yarn.lock with a different host name (https://github.com/containerbuildsystem/cachito/blob/61edce3631ebb4393268e8379760f93d8a07f32b/cachito/workers/pkg_managers/yarn.py#L107).

It should have the same content as the original yarn.lock, but the hostname is replaced.

Reply:
An example Cachito provided yarn.lock:
```
# yarn lockfile v1

"@ardatan/aggregate-error@0.0.1":
  version "0.0.1"
  resolved "https://cachito-nexus.engineering.redhat.com/repository/cachito-yarn-311726/@ardatan/aggregate-error/-/aggregate-error-0.0.1.tgz"
  integrity "sha512-UQ9BequOTIavs0pTHLMwQwKQF8tTV1oezY/H2O9chA+JNPFZSua55xpU5dPSjAU9/jLJ1VwU+HJuTVN8u7S6Fg=="

"@babel/code-frame@^7.0.0":
  version "7.0.0"
  resolved "https://cachito-nexus.engineering.redhat.com/repository/cachito-yarn-311726/@babel/code-frame/-/code-frame-7.0.0.tgz"
  integrity "sha512-OfC2uemaknXr87bdLUkWog7nYuliM9Ij5HUcajsVcMCpQrcLmtxRbVFTIqmcSkSeYRBFBRxs2FiUqFJDLdiebA=="
  dependencies:
    "@babel/highlight" "^7.0.0"

"@babel/code-frame@^7.10.3", "@babel/code-frame@^7.10.4":
  version "7.10.4"
  resolved "https://cachito-nexus.engineering.redhat.com/repository/cachito-yarn-311726/@babel/code-frame/-/code-frame-7.10.4.tgz"
  integrity "sha512-vG6SvB6oYEhvgisZNFRmRCUkLz11c7rp+tbNTynGqc6mS1d5ATd/sGyV6W0KZZnXRKMTzZDRgQT3Ou9jhpAfUg=="
  dependencies:
    "@babel/highlight" "^7.10.4"
```

Reply:
Actually `$REMOTE_SOURCES_DIR/cachito-gomod-with-deps/app/frontend/` also includes the whole frontend source so we can directly build from there. I didn't do that because we want to use build-frontend.sh in the root directory and we may need to run some runtime modifications when copying code to distgit.

Reply:
> It will not be out of sync (unless there's a bug in Cachito). The way that Cachito works is fetching the original source and dependencies then setting up a temporary npm registry for fetched dependencies.

Right, but when does this happen? If the cache doesn't update every time yarn.lock changes, then we could have stale dependencies.

Reply:
It happens on every build.